include/seqan3/range/container/constexpr_string.hpp

```diff
@@ -0,0 +1,595 @@
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr bool empty() const noexcept
+    {
+        return cbegin() == cend();
+    }
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns the begin to the string.
+     *
+     * \returns `iterator` An iterator to the begin of the string.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr iterator begin() noexcept
+    {
+        return &_lit[0];
+    }
+
+    /*!\brief Returns the begin to the string.
+     *
+     * \returns `const_iterator` An iterator to the begin of the string.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr const_iterator begin() const noexcept
+    {
+        return &_lit[0];
+    }
+
+    /*!\brief Returns the begin to the string.
+     *
+     * \returns `const_iterator` An iterator to the begin of the string.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr const_iterator cbegin() const noexcept
+    {
+        return &_lit[0];
+    }
+
+    /*!\brief Returns the end to the string.
+     *
+     * \returns `iterator` An iterator to the end of the string.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr iterator end() noexcept
+    {
+        return &_lit[N];
+    }
+
+    /*!\brief Returns the end to the string.
+     *
+     * \returns `iterator` An iterator to the end of the string.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr const_iterator end() const noexcept
+    {
+        return &_lit[N];
+    }
+
+    /*!\brief Returns the end to the string.
+     *
+     * \returns `const_iterator` An iterator to the end of the string.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr const_iterator cend() const noexcept
+    {
+        return &_lit[N];
+    }
+    //!\}
+
+    /*!\name Operations
+     * \{
+     */
+
+    /*!\brief Concatenates two constexpr_strings.
+     * \param rhs The right-hand-side to concat with.
+     * \returns `constexpr_string<N + N2>` The new constexpr_string with size `N` + `N2`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the strings.
+     */
+    template <size_t N2>
+    constexpr constexpr_string<N + N2> operator+(constexpr_string<N2> const & rhs) const noexcept
+    {
+        return constexpr_string<N + N2>{*this, rhs};
+    }
+
```

Comment:
operator+ is in group Operations, but the comparison operator should be in an extra group, maybe "Comparison operators"?